packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -222,7 +222,7 @@ function mountFragment(
     mountVNodes(children, parent, renderer, anchor);
 
     // children of a fragment will always have at least the two delimiters.
-    vnode.elm = children[children.length - 1]!.elm;
+    vnode.elm = children[0]!.elm;
```

Comment:
Do any of the tests exercise the blocks of code starting on line 882 and line 893, for VFragments? I don't fully understand the implications of this change on the dynamic diffing algo, but those blocks might be impacted. Verifying we have tests for those and they still work would help ease my concerns.

Specifically, that `renderer.nextSibling(oldEndVnode.elm!)` makes me think this change will fix inserting before the VFragment but then break inserting after the VFragment.